src/parser/github-comment-module.ts

```diff
@@ -292,8 +292,10 @@ export class GithubCommentModule extends BaseModule {
       this.context.config.erc20RewardToken
     );
 
-    const rewardsSum = result.comments?.reduce<number>((acc, curr) => acc + (curr.score?.reward ?? 0), 0) ?? 0;
-    const isCapped = result.total < rewardsSum;
+    const rewardsSum =
+      result.comments?.reduce<Decimal>((acc, curr) => acc.add(curr.score?.reward ?? 0), new Decimal(0)) ??
+      new Decimal(0);
+    const isCapped = rewardsSum.gt(taskReward);
```

Comment:
what is this reward cap?

Reply:
This is to limit rewards based on the task price: https://github.com/ubiquity-os-marketplace/text-conversation-rewards/issues/166

Reply:
[[ 0.3 WXDAI ]](https://github.com/Meniole/text-conversation-rewards/issues/undefined) @gentlementlegen
⚠️ Your rewards have been limited to the task price of 0 WXDAI.

This was generated in your QA but it doesn't make sense. 
Why was a reward of 0.3 generated if cap is 0? 
Also if there is no task price I think we shouldn't check for cap

Reply:
Good catch thank you, addressed in https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/209/commits/28d3b8841b6a91baa874b1cf0f1d95a307e296d7

There can be no task reward either if there is no one assigned or no task price.

Reply:
In practice I've never seen that. We don't generate any permits unless the GitHub issue is priced.

Reply:
I think sometimes an issue is about discussing a potential feature but at the end it might get closed because we decided not to implement it ([example](https://github.com/ubiquity/ts-template/issues/67)). I guess everybody that participated in the discussion should get rewarded, maybe not as much as if it gets implemented but everybody still put time into it.

---

@gentlementlegen as I understand the cap is only for users that are not assigned to the task, so this should also check that?

Reply:
The reason why I check for the task itself is because we have a scenario where you can have multiple assignees to a task, so the actual reward gets split (e.g. 100 USD task will be rewarded 50 USD if there are two users assigned) so the cap would be 50 USD, not 100. If there is no assignee I will fall back to the task price if there is any, I will change that.

---

@whilefoo the limitation applies also for the assignees, that cannot have comment / spec rewards above the task reward either.

Reply:
This is an interesting scenario that I didn't think of but actually the cap should remain at whatever the label is for all the users. I think it makes sense that if two assignees are splitting the reward that it shouldn't penalize the others. Only one could possibly have been the author. I suppose there is a slim chance that one can theoretically farm comment rewards while not being assigned and then be assigned at the very end. I suppose it would still be considered valid? I'm not sure a good solution for this. 

Also for whatever it's worth, right before closing out an issue a super easy way to troll would be to team assign yourself across multiple accounts knocking down rewards for everybody else. Anyways I think let's retain caps at the price label.

Reply:
@0x4007 All right then let me change the cap rewards to the `Price: ---USD` tag label assigned to that task, so every user gets limited to that task price.